src/util/bbox.ts

```diff
@@ -56,86 +67,500 @@ export const from = (bboxes: BBox[]): BBox => {
+        }
+      },
+      get top() {
+        return untrack(() => {
+          if ("top" in equations.y) {
+            return equations.y.top![1];
+          } else if (centerYAndHeight() === undefined) {
+            return undefined;
+          } else {
+            return centerYAndHeight()![0] - centerYAndHeight()![1] / 2;
+          }
+        });
+      },
+      set top(top: number | undefined) {
+        if (top === undefined) {
+          setEquations(
+            "y",
+            produce((dims) => {
+              delete dims.top;
+            })
+          );
+        } else {
+          setEquations("y", "top", [[1, -0.5], top]);
+        }
+      },
+      get centerY() {
+        return untrack(() => {
+          if ("centerY" in equations.y) {
+            return equations.y.centerY![1];
+          } else if (centerYAndHeight() === undefined) {
+            return undefined;
+          } else {
+            return centerYAndHeight()![0];
+          }
+        });
+      },
+      set centerY(centerY: number | undefined) {
+        if (centerY === undefined) {
+          setEquations(
+            "y",
+            produce((dims) => {
+              delete dims.centerY;
+            })
+          );
+        } else {
+          setEquations("y", "centerY", [[1, 0], centerY]);
+        }
+      },
+      get bottom() {
+        return untrack(() => {
+          if ("bottom" in equations.y) {
+            return equations.y.bottom![1];
+          } else if (centerYAndHeight() === undefined) {
+            return undefined;
+          } else {
+            return centerYAndHeight()![0] + centerYAndHeight()![1] / 2;
+          }
+        });
+      },
+      set bottom(bottom: number | undefined) {
+        if (bottom === undefined) {
+          setEquations(
+            "y",
+            produce((dims) => {
+              delete dims.bottom;
+            })
+          );
+        } else {
+          setEquations("y", "bottom", [[1, 0.5], bottom]);
+        }
+      },
+      get height() {
+        return untrack(() => {
+          if ("height" in equations.y) {
+            return equations.y.height![1];
+          } else if (centerYAndHeight() === undefined) {
+            return undefined;
+          } else {
+            return centerYAndHeight()![1];
+          }
+        });
+      },
+      set height(height: number | undefined) {
+        if (height === undefined) {
+          setEquations(
+            "y",
+            produce((dims) => {
+              delete dims.height;
+            })
+          );
+        } else {
+          setEquations("y", "height", [[0, 1], height]);
+        }
+      },
+    },
+    owners: {
+      get left() {
+        return untrack(() => {
+          if ("left" in owners.x) {
+            return owners.x.left;
+          } else if (Object.keys(equations.x).length === 2) {
+            return inferred;
+          } else {
+            return undefined;
+          }
```

Comment:
I added some commentary to the top-level comment. I could refactor everything into a function, but I'm choosing to keep it inline for now. May revisit in the future

Reply:
cool, thats fine with me. Here's some extra feedback you can take or leave:

I understand now the role of inferred in this file, but not what its value is or how it ends up being used. I looked up the definition of `inferred` in Scenegraph.ts:
```
export type Inferred = { inferred: true };
export const inferred: Inferred = { inferred: true };
```
I can now infer (pun unintended) that inferred is a flag that bluefish uses to know it can calculate a property based on other properties. I wouldn't know where to look up where Bluefish does that though. If I wasn't being lazy I could search the while project for `inferred` and see where it gets checked... but I guess I'm thinking it could be useful to have a comment next to the definition of inferred in Scenegraph.ts that says where it gets used / checked